interfaces/tls_certificates/v1/README.md

```diff
@@ -18,6 +26,7 @@ compatible with the interface.
 
 - Is expected to generate (or use an existing) private-key
 - Is expected to provide a list of CSR's for which it requires certificates
+- Is expected to specify whether the certificate request is for a Certificate Authority (CA) or not
```

Comment:
I do not think this is needed, as the CSR could be provided with the `KeyUsage` extension to show it is for a CA certificate.

Reply:
Unfortunately this is not specified in the CSR, it's set on certificate generation time.

For example when Vault is set as an intermediate CA, it gives you a CSR and there's nothing in it that tells you that it's for a CA.

Reply:
Ok, it is possible to specifically add the extension in the CSR, and that would have been cleaner, but since we have some use cases where the extension is not present, I don't see a better alternative.

Reply:
Vault does have a way to add Basic Constraints extension with CA: true in the CSR but I'm not sure how standard is this.

```
        Attributes:
            Requested Extensions:
                X509v3 Subject Alternative Name: 
                    DNS:pizza.com
                X509v3 Basic Constraints: critical
                    CA:TRUE
```

Reference:
- https://developer.hashicorp.com/vault/api-docs/secret/pki#parameters-20